source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.md

```diff
@@ -633,6 +751,11 @@ if connection state is "pending":
     decrement pendingConnectionCount
 else:
     decrement availableConnectionCount
+
+error = await_pending_response(pool, connection)
+if error:
+  return error
```

Comment:
We should fix the pseudocode by emitting the `ConnectionCheckOutFailedEvent` before throwing this `error`. This issue was identified in https://github.com/mongodb/specifications/pull/1675#discussion_r2337951642, I am just pointing out to at least one place where the change has to be done.